System.IO.Abstractions/IFileStreamFactory.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Security.AccessControl;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Abstractions
+{
+    public interface IFileStreamFactory
+    {
+        Stream Create(string path, FileMode mode);
+
+        Stream Create(string path, FileMode mode, FileAccess access);
+        
+        Stream Create(string path, FileMode mode, FileAccess access, FileShare share);
+        
+        Stream Create(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize);
+        
+        Stream Create(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options);
+        
+        Stream Create(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, bool useAsync);
+
+        Stream Create(SafeFileHandle handle, FileAccess access);
+        
+        Stream Create(SafeFileHandle handle, FileAccess access, int bufferSize);
+        
+        Stream Create(SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync);
+
+#if NET40
+        Stream Create(string path, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity);
+
+        Stream Create(string path, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options);
+#endif
+
+#if NET40 || NETSTANDARD_20
```

Comment:
Do we need the NET40/NETSTANDARD check here? I don't see a mirroring check in the MockFileStreamFactory

Reply:
Yeah, because the “real” implementation doesn’t support these overloads on NS1.4.